e2e_testing/fixtures/002_carousel.sql

```diff
@@ -0,0 +1,43 @@
+INSERT INTO learning_resources_learningresourceofferor (created_on, updated_on, code, name, professional)
+SELECT (data ->> 'created_on')::timestamptz AS created_on,
+       (data ->> 'updated_on')::timestamptz AS updated_on,
+       (data ->> 'code')::text AS code,
+       (data ->> 'name')::text AS name,
+       (data ->> 'professional')::boolean AS professional
+FROM jsonb_array_elements('[
+  {
+    "created_on": "2024-01-01",
+    "updated_on": "2024-01-01",
+    "code": "mitx",
+    "name": "MITx",
+    "professional": false
+  }
+]'::jsonb) AS item(data);
+
+INSERT INTO learning_resources_learningresource (id, created_on, updated_on, readable_id, title, description, published, resource_type, etl_source, professional, offered_by_id)
+SELECT (data ->> 'id')::integer AS id,
+       (data ->> 'created_on')::timestamptz AS created_on,
+       (data ->> 'updated_on')::timestamptz AS updated_on,
+       (data ->> 'readable_id')::text AS readable_id,
+       (data ->> 'title')::text AS title,
+       (data ->> 'description')::text AS description,
+       (data ->> 'published')::boolean AS published,
+       (data ->> 'resource_type')::text AS resource_type,
+       (data ->> 'etl_source')::text AS etl_source,
+       (data ->> 'professional')::boolean AS professional,
+       (data ->> 'offered_by_id')::text AS offered_by_id
+FROM jsonb_array_elements('[
+  {
+    "id": 100000,
+    "created_on": "2024-01-01",
+    "updated_on": "2024-01-01",
+    "readable_id": "test-100000",
+    "title": "Test 100000",
+    "description": "Description",
+    "published": true,
+    "resource_type": "program",
+    "etl_source": "micromasters",
+    "professional": false,
+    "offered_by_id": "mitx"
+  }
+]'::jsonb) AS item(data);
```

Comment:
Instead of sql, maybe the `loaddata` command could be used instead?  For example `manage.py loaddata platforms departments offered_by` upserts data stored in JSON files from `learning_resources/fixtures` folder.

Reply:
Where we can, I think we should create test data via the APIs. Once the e2e tests have authentication worked out, that should work for things like learning paths or user lists. (Tests that are actually testing the UI creation would create via UI not via APIs.) But many of our core data models have read-only APIs since the data is ingested from other platforms—e.g., courses, programs, podcasts. 

> The idea is that the tests should have as little coupling as possible to any service and so was avoiding any dependency on Django. In practical terms though I'm not sure if there's much benefit - perhaps future scenario where we want to apply data on environments where the manage.py tool isn't available.

My 2c: Anywhere we have direct access to the DB for running SQL, then we probably have access to django / `./manage.py` also. Switching away from Django doesn't seem likely, and if we ever did, converting some data loading to another format seems like a small lift compared to everything else.

So I think we should use `loaddata` **if it is more convenient**. Seems to me that the primary advantage is we can store the data as just JSON rather than as a SQL query.

**If we keep the SQL:** Then @jonkafton a slightly more convenient form of the json insertion command is
```sql
INSERT INTO learning_resources_learningresource
SELECT *
FROM jsonb_populate_recordset(null::learning_resources_learningresource, '[
  -- obj1,
  -- obj2,
  -- etc
]')
```
The first argument to `json_populate_recordset` is an object specifying the shape of the data. https://www.postgresql.org/docs/current/functions-json.html

Reply:
> Where we can, I think we should create test data via the APIs. Once the e2e tests have authentication worked out, that should work for things like learning paths or user lists. (Tests that are actually testing the UI creation would create via UI not via APIs.) But many of our core data models have read-only APIs since the data is ingested from other platforms—e.g., courses, programs, podcasts.

I've added a note here that aims to capture this: https://github.com/mitodl/mit-open/blob/jk/418-e2e-tests-ci/e2e_testing/README.md?plain=1#L73. We can certainly use the APIs and potentially use this suite to test APIs directly. We might want to be careful mixing API and UI tests in case we ever want to port to a dedicated integration test suite.

Reply:
> So I think we should use loaddata if it is more convenient. Seems to me that the primary advantage is we can store the data as just JSON rather than as a SQL query.

I've switched to using Django fixtures in https://github.com/mitodl/mit-open/commit/d69d8038 and it does feel much cleaner. I was thinking of a scenario where we are testing a live environment, RC, and want to apply SQL directly, though I guess if we are allowed to access the DB, we are also allowed to access the web container.